src/webui/webapplication.cpp

```diff
@@ -49,101 +52,21 @@
-const QString SCOPE_IMAGES = "images";
-const QString SCOPE_THEME = "theme";
-const QString DEFAULT_ACTION = "index";
-const QString WEBUI_ACTION = "webui";
-const QString VERSION_INFO = "version";
-const QString MAX_AGE_MONTH = "public, max-age=2592000";
-
-#define ADD_ACTION(scope, action) actions[#scope][#action] = &WebApplication::action_##scope##_##action
-
-QMap<QString, QMap<QString, WebApplication::Action>> WebApplication::initializeActions()
-{
-    QMap<QString,QMap<QString, WebApplication::Action>> actions;
-
-    ADD_ACTION(public, webui);
-    ADD_ACTION(public, index);
-    ADD_ACTION(public, login);
-    ADD_ACTION(public, logout);
-    ADD_ACTION(public, theme);
-    ADD_ACTION(public, images);
-    ADD_ACTION(query, torrents);
-    ADD_ACTION(query, preferences);
-    ADD_ACTION(query, transferInfo);
-    ADD_ACTION(query, propertiesGeneral);
-    ADD_ACTION(query, propertiesTrackers);
-    ADD_ACTION(query, propertiesWebSeeds);
-    ADD_ACTION(query, propertiesFiles);
-    ADD_ACTION(query, getLog);
-    ADD_ACTION(query, getPeerLog);
-    ADD_ACTION(query, getPieceHashes);
-    ADD_ACTION(query, getPieceStates);
-    ADD_ACTION(sync, maindata);
-    ADD_ACTION(sync, torrent_peers);
-    ADD_ACTION(command, shutdown);
-    ADD_ACTION(command, download);
-    ADD_ACTION(command, upload);
-    ADD_ACTION(command, addTrackers);
-    ADD_ACTION(command, resumeAll);
-    ADD_ACTION(command, pauseAll);
-    ADD_ACTION(command, resume);
-    ADD_ACTION(command, pause);
-    ADD_ACTION(command, setPreferences);
-    ADD_ACTION(command, setFilePrio);
-    ADD_ACTION(command, getGlobalUpLimit);
-    ADD_ACTION(command, getGlobalDlLimit);
-    ADD_ACTION(command, setGlobalUpLimit);
-    ADD_ACTION(command, setGlobalDlLimit);
-    ADD_ACTION(command, getTorrentsUpLimit);
-    ADD_ACTION(command, getTorrentsDlLimit);
-    ADD_ACTION(command, setTorrentsUpLimit);
-    ADD_ACTION(command, setTorrentsDlLimit);
-    ADD_ACTION(command, alternativeSpeedLimitsEnabled);
-    ADD_ACTION(command, toggleAlternativeSpeedLimits);
-    ADD_ACTION(command, toggleSequentialDownload);
-    ADD_ACTION(command, toggleFirstLastPiecePrio);
-    ADD_ACTION(command, setSuperSeeding);
-    ADD_ACTION(command, setForceStart);
-    ADD_ACTION(command, delete);
-    ADD_ACTION(command, deletePerm);
-    ADD_ACTION(command, increasePrio);
-    ADD_ACTION(command, decreasePrio);
-    ADD_ACTION(command, topPrio);
-    ADD_ACTION(command, bottomPrio);
-    ADD_ACTION(command, setLocation);
-    ADD_ACTION(command, rename);
-    ADD_ACTION(command, setAutoTMM);
-    ADD_ACTION(command, recheck);
-    ADD_ACTION(command, setCategory);
-    ADD_ACTION(command, addCategory);
-    ADD_ACTION(command, removeCategories);
-    ADD_ACTION(command, getSavePath);
-    ADD_ACTION(version, api);
-    ADD_ACTION(version, api_min);
-    ADD_ACTION(version, qbittorrent);
-
-    return actions;
-}
+constexpr Utils::Version<int, 3, 2> API_VERSION {2, 0, 0};
```

Comment:
When converted to QString this will print `2.0.0` right?
If so, then it will break all existing usage/detection.

Reply:
Well, the points of incompatibility creation should be it the development process, otherwise we will always have to live with the mistakes previously...
Really, there should not be a big problem since it a major update and old clients can't work with it anyway (because of changed API method names/paths).

Reply:
>Really, there should not be a big problem since it a major update and old clients can't work with it anyway (because of changed API method names/paths).

OK, then we **should** inform the public the changes beforehand, making an announcement on the qbt website news page when this PR is merged, and this should be as early as possible as we have a lot of users using webAPI.
@sledgehammer999

Reply:
@glassez 
I was thinking this should start following [Semantic Versioning](http://semver.org/).
If so, this should go `{old API_VERSION + 1, 0, 0};`, what do you think?

Reply:
I follow it but in another way. I try to follow qBittorrent main versioning rule when we change major version only on some very big changes. So I interpret old version as 1.x.y in new scheme and just perform major update on it.
But if most of you guys prefer "to be measured penises" (as Firefox and Chrome do), then I'll change it to the option you proposed.

Reply:
* Using Semantic Versioning is one thing, we can start practicing it if the majority agrees.
  IMO Semantic Versioning is invented just for this situation.
* The `{2, 0, 0};` you proposed is another, problem is: with the current `/version/api GET` it returns an integer `15`, but after this PR it drops back to `2.0.0` which is a confusion for our users.
  So from a user perspective I would recommend using `16.0.0`.
  Also why not use `Utils::Version<int, 3, 3>`?
  https://github.com/qbittorrent/qBittorrent/wiki/WebUI-API-Documentation#get-api-version

Reply:
>it returns an integer 15, but after this PR it drops back to 2.0.0 which is a confusion for our users.

It's not a problem, IMO. But I'll do whatever the team decides.
>Also why not use Utils::Version<int, 3, 3>?

Personally I prefer the way when zeros after minor version are truncated. Waiting other reviewers.

Reply:
@lgallard 
what do you think about the suggestion in: https://github.com/qbittorrent/qBittorrent/pull/7610/files#r145924980